Tests/StreamChatTests/Controllers/CurrentUserController/CurrentUserController_Tests.swift

```diff
@@ -412,11 +445,44 @@ final class CurrentUserController_Tests: XCTestCase {
         XCTAssert(error is ClientError.CurrentUserDoesNotExist)
     }
     
+    func test_synchronizeDevices_doesNotDeadlock() throws {
+        // GIVEN
+        // Simulate saving current user to a database
+        try client.databaseContainer.writeSynchronously {
+            let currentUserPayload: CurrentUserPayload = .dummy(
+                userId: .unique,
+                role: .user,
+                unreadCount: .none
+            )
+            try $0.saveCurrentUser(payload: currentUserPayload)
+        }
+        
+        // WHEN
+        // updateUser is called from background queue
+        DispatchQueue.global().async {
+            self.controller.synchronizeDevices()
+        }
+        
+        // Sleep main thread for a little time
+        // to make sure background thread acquires lock (from Atomic, in EntityDatabaseObserver.item)
+        // if we don't sleep, main thread acquires lock first & no deadlock occurs
+        sleep(1)
+        
+        // THEN
+        // updateUser is called from main queue and actually finishes
+        let exp = expectation(description: "completion called")
+        controller.synchronizeDevices() { _ in
+            exp.fulfill()
+        }
+        
+        wait(for: [exp], timeout: 5)
```

Comment:
```suggestion
        wait(for: [exp], timeout: defaultTimeout)
```

Reply:
> Termination Reason: FRONTBOARD 2343432205 
<RBSTerminateContext| domain:10 code:0x8BADF00D explanation:[application<com.zumper.Rentals-test>:49789] failed to terminate gracefully after 5.0s

Real timeout for a deadlock is 5s, i'd say we should keep it as 5, wdyt?